### ISSUE TITLE
[Bug] Unable to read game versions properly

### ISSUE THREAD

**Issue by zhenlige** [Kir-Antipov/mc-publish#135]
### Description

```Markdown
The action fails no matter how `game-versions` is set. The corresponding field in `fabric.mod.json` is already set to `">=1.21.2-"`.
```

### Expected Behavior

The mod should be successfully built and published.

### Actual Behavior

The action always fails.

- Left empty or set to `">=1.21.2-"`: `ArgumentError: At least one game version should be specified to upload files to Modrinth. (Parameter 'request.gameVersions')`
- Set to `[1.21.2, 1.21.3]`: `The workflow is not valid. .github/workflows/publish.yml (Line: 33, Col: 26): A sequence was not expected`

### Version

3.3.0

### Environment

GitHub Actions

### Configuration

```yml
name: publish
on: [workflow_dispatch]

permissions:
  contents: write

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - name: checkout
        uses: actions/checkout@v4
        with:
          submodules: true
      - name: setup Java
        uses: actions/setup-java@v4
        with:
          java-version: 21
          distribution: 'microsoft'
      - name: exe
        if: ${{ runner.os != 'Windows' }}
        run: chmod +x ./gradlew
      - name: build
        run: ./gradlew clean build
      - name: release
        uses: Kir-Antipov/mc-publish@v3.3
        with:
          modrinth-id: xennote
          modrinth-token: ${{ secrets.MODRINTH_TOKEN }}

          github-token: ${{ secrets.GAYHUB_TOKEN }}

          game-versions: [1.21.2, 1.21.3]

          version-type: beta
```

### Logs

```sh
2024-11-28T07:47:27.3591022Z Current runner version: '2.321.0'
2024-11-28T07:47:27.3616986Z ##[group]Operating System
2024-11-28T07:47:27.3617782Z Ubuntu
2024-11-28T07:47:27.3618383Z 22.04.5
2024-11-28T07:47:27.3618893Z LTS
2024-11-28T07:47:27.3619352Z ##[endgroup]
2024-11-28T07:47:27.3619968Z ##[group]Runner Image
2024-11-28T07:47:27.3620531Z Image: ubuntu-22.04
2024-11-28T07:47:27.3621325Z Version: 20241124.1.0
2024-11-28T07:47:27.3622478Z Included Software: https://github.com/actions/runner-images/blob/ubuntu22/20241124.1/images/ubuntu/Ubuntu2204-Readme.md
2024-11-28T07:47:27.3623865Z Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20241124.1
2024-11-28T07:47:27.3624825Z ##[endgroup]
2024-11-28T07:47:27.3625376Z ##[group]Runner Image Provisioner
2024-11-28T07:47:27.3625981Z 2.0.385.1
2024-11-28T07:47:27.3626830Z ##[endgroup]
2024-11-28T07:47:27.3627873Z ##[group]GITHUB_TOKEN Permissions
2024-11-28T07:47:27.3630014Z Contents: write
2024-11-28T07:47:27.3630587Z Metadata: read
2024-11-28T07:47:27.3631288Z ##[endgroup]
2024-11-28T07:47:27.3634578Z Secret source: Actions
2024-11-28T07:47:27.3635694Z Prepare workflow directory
2024-11-28T07:47:27.3941200Z Prepare all required actions
2024-11-28T07:47:27.3977713Z Getting action download info
2024-11-28T07:47:27.6082865Z Download action repository 'actions/checkout@v4' (SHA:11bd71901bbe5b1630ceea73d27597364c9af683)
2024-11-28T07:47:27.7048418Z Download action repository 'actions/setup-java@v4' (SHA:8df1039502a15bceb9433410b1a100fbe190c53b)
2024-11-28T07:47:27.9604142Z Download action repository 'Kir-Antipov/mc-publish@v3.3' (SHA:995edadc13559a8b28d0b7e6571229f067ec7659)
2024-11-28T07:47:28.5442210Z Complete job name: build
2024-11-28T07:47:28.6173161Z ##[group]Run actions/checkout@v4
2024-11-28T07:47:28.6174339Z with:
2024-11-28T07:47:28.6174754Z   submodules: true
2024-11-28T07:47:28.6175278Z   repository: zhenlige/xennote
2024-11-28T07:47:28.6176007Z   token: ***
2024-11-28T07:47:28.6176407Z   ssh-strict: true
2024-11-28T07:47:28.6176917Z   ssh-user: git
2024-11-28T07:47:28.6177394Z   persist-credentials: true
2024-11-28T07:47:28.6177917Z   clean: true
2024-11-28T07:47:28.6178468Z   sparse-checkout-cone-mode: true
2024-11-28T07:47:28.6179019Z   fetch-depth: 1
2024-11-28T07:47:28.6179482Z   fetch-tags: false
2024-11-28T07:47:28.6179983Z   show-progress: true
2024-11-28T07:47:28.6180399Z   lfs: false
2024-11-28T07:47:28.6180865Z   set-safe-directory: true
2024-11-28T07:47:28.6181651Z ##[endgroup]
2024-11-28T07:47:28.8306569Z Syncing repository: zhenlige/xennote
2024-11-28T07:47:28.8309803Z ##[group]Getting Git version info
2024-11-28T07:47:28.8310857Z Working directory is '/home/runner/work/xennote/xennote'
2024-11-28T07:47:28.8312394Z [command]/usr/bin/git version
2024-11-28T07:47:28.8365364Z git version 2.47.0
2024-11-28T07:47:28.8395034Z ##[endgroup]
2024-11-28T07:47:28.8416749Z Temporarily overriding HOME='/home/runner/work/_temp/6797b18f-b13c-4f9f-859d-b56074f2a9cc' before making global git config changes
2024-11-28T07:47:28.8419474Z Adding repository directory to the temporary git global config as a safe directory
2024-11-28T07:47:28.8423544Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/xennote/xennote
2024-11-28T07:47:28.8463255Z Deleting the contents of '/home/runner/work/xennote/xennote'
2024-11-28T07:47:28.8467728Z ##[group]Initializing the repository
2024-11-28T07:47:28.8472279Z [command]/usr/bin/git init /home/runner/work/xennote/xennote
2024-11-28T07:47:28.8562295Z hint: Using 'master' as the name for the initial branch. This default branch name
2024-11-28T07:47:28.8563843Z hint: is subject to change. To configure the initial branch name to use in all
2024-11-28T07:47:28.8564994Z hint: of your new repositories, which will suppress this warning, call:
2024-11-28T07:47:28.8565888Z hint:
2024-11-28T07:47:28.8566860Z hint: 	git config --global init.defaultBranch <name>
2024-11-28T07:47:28.8567836Z hint:
2024-11-28T07:47:28.8568768Z hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
2024-11-28T07:47:28.8570586Z hint: 'development'. The just-created branch can be renamed via this command:
2024-11-28T07:47:28.8572436Z hint:
2024-11-28T07:47:28.8573535Z hint: 	git branch -m <name>
2024-11-28T07:47:28.8574945Z Initialized empty Git repository in /home/runner/work/xennote/xennote/.git/
2024-11-28T07:47:28.8582148Z [command]/usr/bin/git remote add origin https://github.com/zhenlige/xennote
2024-11-28T07:47:28.8619732Z ##[endgroup]
2024-11-28T07:47:28.8621157Z ##[group]Disabling automatic garbage collection
2024-11-28T07:47:28.8624703Z [command]/usr/bin/git config --local gc.auto 0
2024-11-28T07:47:28.8655293Z ##[endgroup]
2024-11-28T07:47:28.8656859Z ##[group]Setting up auth
2024-11-28T07:47:28.8663421Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2024-11-28T07:47:28.8695141Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2024-11-28T07:47:28.9006687Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2024-11-28T07:47:28.9039221Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2024-11-28T07:47:28.9280762Z [command]/usr/bin/git config --local http.https://github.com/.extraheader AUTHORIZATION: basic ***
2024-11-28T07:47:28.9318614Z ##[endgroup]
2024-11-28T07:47:28.9320236Z ##[group]Fetching the repository
2024-11-28T07:47:28.9328791Z [command]/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --no-recurse-submodules --depth=1 origin +b49119df2d913144183bca65767f4e97e3110c79:refs/remotes/origin/main
2024-11-28T07:47:29.3682528Z From https://github.com/zhenlige/xennote
2024-11-28T07:47:29.3685028Z  * [new ref]         b49119df2d913144183bca65767f4e97e3110c79 -> origin/main
2024-11-28T07:47:29.3726461Z ##[endgroup]
2024-11-28T07:47:29.3728979Z ##[group]Determining the checkout info
2024-11-28T07:47:29.3731906Z ##[endgroup]
2024-11-28T07:47:29.3733583Z [command]/usr/bin/git sparse-checkout disable
2024-11-28T07:47:29.3775960Z [command]/usr/bin/git config --local --unset-all extensions.worktreeConfig
2024-11-28T07:47:29.3809547Z ##[group]Checking out the ref
2024-11-28T07:47:29.3812520Z [command]/usr/bin/git checkout --progress --force -B main refs/remotes/origin/main
2024-11-28T07:47:29.4250541Z Switched to a new branch 'main'
2024-11-28T07:47:29.4252970Z branch 'main' set up to track 'origin/main'.
2024-11-28T07:47:29.4260277Z ##[endgroup]
2024-11-28T07:47:29.4262284Z ##[group]Setting up auth for fetching submodules
2024-11-28T07:47:29.4265587Z [command]/usr/bin/git config --global http.https://github.com/.extraheader AUTHORIZATION: basic ***
2024-11-28T07:47:29.4304604Z [command]/usr/bin/git config --global --unset-all url.https://github.com/.insteadOf
2024-11-28T07:47:29.4337198Z [command]/usr/bin/git config --global --add url.https://github.com/.insteadOf git@github.com:
2024-11-28T07:47:29.4370742Z [command]/usr/bin/git config --global --add url.https://github.com/.insteadOf org-38127204@github.com:
2024-11-28T07:47:29.4401306Z ##[endgroup]
2024-11-28T07:47:29.4404540Z ##[group]Fetching submodules
2024-11-28T07:47:29.4406657Z [command]/usr/bin/git submodule sync
2024-11-28T07:47:29.4644256Z [command]/usr/bin/git -c protocol.version=2 submodule update --init --force --depth=1
2024-11-28T07:47:29.4904921Z [command]/usr/bin/git submodule foreach git config --local gc.auto 0
2024-11-28T07:47:29.5177391Z ##[endgroup]
2024-11-28T07:47:29.5179946Z ##[group]Persisting credentials for submodules
2024-11-28T07:47:29.5185006Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'url\.https\:\/\/github\.com\/\.insteadOf' && git config --local --unset-all 'url.https://github.com/.insteadOf' || :"
2024-11-28T07:47:29.5421193Z [command]/usr/bin/git submodule foreach sh -c "git config --local 'http.https://github.com/.extraheader' 'AUTHORIZATION: basic ***' && git config --local --show-origin --name-only --get-regexp remote.origin.url"
2024-11-28T07:47:29.5660172Z [command]/usr/bin/git submodule foreach git config --local --add 'url.https://github.com/.insteadOf' 'git@github.com:'
2024-11-28T07:47:29.5895794Z [command]/usr/bin/git submodule foreach git config --local --add 'url.https://github.com/.insteadOf' 'org-38127204@github.com:'
2024-11-28T07:47:29.6128853Z ##[endgroup]
2024-11-28T07:47:29.6165966Z [command]/usr/bin/git log -1 --format=%H
2024-11-28T07:47:29.6190420Z b49119df2d913144183bca65767f4e97e3110c79
2024-11-28T07:47:29.6597302Z ##[group]Run actions/setup-java@v4
2024-11-28T07:47:29.6598617Z with:
2024-11-28T07:47:29.6599570Z   java-version: 21
2024-11-28T07:47:29.6600511Z   distribution: microsoft
2024-11-28T07:47:29.6601570Z   java-package: jdk
2024-11-28T07:47:29.6602592Z   check-latest: false
2024-11-28T07:47:29.6603588Z   server-id: github
2024-11-28T07:47:29.6604535Z   server-username: GITHUB_ACTOR
2024-11-28T07:47:29.6605919Z   server-password: GITHUB_TOKEN
2024-11-28T07:47:29.6607080Z   overwrite-settings: true
2024-11-28T07:47:29.6608136Z   job-status: success
2024-11-28T07:47:29.6609532Z   token: ***
2024-11-28T07:47:29.6610428Z ##[endgroup]
2024-11-28T07:47:29.8563163Z ##[group]Installed distributions
2024-11-28T07:47:29.8588609Z Trying to resolve the latest version from remote
2024-11-28T07:47:30.0415048Z Resolved latest version as 21.0.2
2024-11-28T07:47:30.0415965Z Trying to download...
2024-11-28T07:47:30.0418145Z Downloading Java 21.0.2 (Microsoft) from https://aka.ms/download-jdk/microsoft-jdk-21.0.2-linux-x64.tar.gz ...
2024-11-28T07:47:31.8293165Z Extracting Java archive...
2024-11-28T07:47:31.8405881Z [command]/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/8994dc94-4fa1-4d4b-a53a-f1e8f36d4313 -f /home/runner/work/_temp/f456e6e3-4ccc-400e-bdfb-80da2203552a
2024-11-28T07:47:34.7646644Z Java 21.0.2 was downloaded
2024-11-28T07:47:34.7647409Z Setting Java 21.0.2 as the default
2024-11-28T07:47:34.7660335Z Creating toolchains.xml for JDK version 21 from microsoft
2024-11-28T07:47:34.7741621Z Writing to /home/runner/.m2/toolchains.xml
2024-11-28T07:47:34.7742442Z 
2024-11-28T07:47:34.7742760Z Java configuration:
2024-11-28T07:47:34.7743443Z   Distribution: microsoft
2024-11-28T07:47:34.7744120Z   Version: 21.0.2
2024-11-28T07:47:34.7745094Z   Path: /opt/hostedtoolcache/Java_Microsoft_jdk/21.0.2/x64
2024-11-28T07:47:34.7745669Z 
2024-11-28T07:47:34.7746771Z ##[endgroup]
2024-11-28T07:47:34.7767326Z Creating settings.xml with server-id: github
2024-11-28T07:47:34.7768261Z Writing to /home/runner/.m2/settings.xml
2024-11-28T07:47:34.7922443Z ##[group]Run chmod +x ./gradlew
2024-11-28T07:47:34.7922959Z [36;1mchmod +x ./gradlew[0m
2024-11-28T07:47:34.8033738Z shell: /usr/bin/bash -e {0}
2024-11-28T07:47:34.8034113Z env:
2024-11-28T07:47:34.8034675Z   JAVA_HOME: /opt/hostedtoolcache/Java_Microsoft_jdk/21.0.2/x64
2024-11-28T07:47:34.8035325Z   JAVA_HOME_21_X64: /opt/hostedtoolcache/Java_Microsoft_jdk/21.0.2/x64
2024-11-28T07:47:34.8035828Z ##[endgroup]
2024-11-28T07:47:34.8185030Z ##[group]Run ./gradlew clean build
2024-11-28T07:47:34.8185608Z [36;1m./gradlew clean build[0m
2024-11-28T07:47:34.8242256Z shell: /usr/bin/bash -e {0}
2024-11-28T07:47:34.8242642Z env:
2024-11-28T07:47:34.8243038Z   JAVA_HOME: /opt/hostedtoolcache/Java_Microsoft_jdk/21.0.2/x64
2024-11-28T07:47:34.8243643Z   JAVA_HOME_21_X64: /opt/hostedtoolcache/Java_Microsoft_jdk/21.0.2/x64
2024-11-28T07:47:34.8244150Z ##[endgroup]
2024-11-28T07:47:34.9227165Z Downloading https://services.gradle.org/distributions/gradle-8.8-bin.zip
2024-11-28T07:47:36.2315912Z .............10%.............20%.............30%.............40%.............50%.............60%..............70%.............80%.............90%.............100%
2024-11-28T07:47:37.4472799Z 
2024-11-28T07:47:37.4479086Z Welcome to Gradle 8.8!
2024-11-28T07:47:37.4479466Z 
2024-11-28T07:47:37.4479708Z Here are the highlights of this release:
2024-11-28T07:47:37.4480824Z  - Running Gradle on Java 22
2024-11-28T07:47:37.4520094Z  - Configurable Gradle daemon JVM
2024-11-28T07:47:37.4520623Z  - Improved IDE performance for large projects
2024-11-28T07:47:37.4521625Z 
2024-11-28T07:47:37.4521942Z For more details see https://docs.gradle.org/8.8/release-notes.html
2024-11-28T07:47:37.4522383Z 
2024-11-28T07:47:37.5466333Z Starting a Gradle Daemon (subsequent builds will be faster)
2024-11-28T07:47:56.3470586Z 
2024-11-28T07:47:56.3500827Z > Configure project :
2024-11-28T07:47:56.3505923Z Fabric Loom: 1.7.4
2024-11-28T07:48:22.7468073Z :remapping 49 mods from modImplementation (java-api)
2024-11-28T07:48:24.7465980Z :remapping 3 mods from modApi (java-api)
2024-11-28T07:48:26.4464706Z 
2024-11-28T07:48:26.4467199Z > Task :clean
2024-11-28T07:48:56.4465496Z > Task :compileJava
2024-11-28T07:48:59.8467464Z > Task :processResources
2024-11-28T07:48:59.8468441Z > Task :classes
2024-11-28T07:49:02.0466555Z > Task :compileClientJava
2024-11-28T07:49:02.0467472Z > Task :processClientResources
2024-11-28T07:49:02.0467917Z > Task :jar
2024-11-28T07:49:02.1464871Z 
2024-11-28T07:49:02.1465829Z > Task :processIncludeJars
2024-11-28T07:49:02.1466829Z (1.2) is not valid semver for dependency org.apache.commons:commons-numbers-primes:1.2
2024-11-28T07:49:02.1467809Z (4.0-beta1) is not valid semver for dependency org.apache.commons:commons-math4-core:4.0-beta1
2024-11-28T07:49:03.6475493Z 
2024-11-28T07:49:03.6475943Z > Task :sourcesJar
2024-11-28T07:49:03.7465599Z > Task :compileTestJava NO-SOURCE
2024-11-28T07:49:03.7466701Z > Task :processTestResources NO-SOURCE
2024-11-28T07:49:03.7467383Z > Task :testClasses UP-TO-DATE
2024-11-28T07:49:03.7468324Z > Task :test NO-SOURCE
2024-11-28T07:49:04.6465597Z > Task :validateAccessWidener
2024-11-28T07:49:04.6466690Z > Task :check
2024-11-28T07:49:05.4465576Z > Task :remapJar
2024-11-28T07:49:06.6465389Z > Task :remapSourcesJar
2024-11-28T07:49:06.6465967Z > Task :assemble
2024-11-28T07:49:06.6468428Z > Task :build
2024-11-28T07:49:06.6468717Z 
2024-11-28T07:49:06.6468886Z BUILD SUCCESSFUL in 1m 31s
2024-11-28T07:49:06.6469307Z 11 actionable tasks: 11 executed
2024-11-28T07:49:07.0013614Z ##[group]Run Kir-Antipov/mc-publish@v3.3
2024-11-28T07:49:07.0013999Z with:
2024-11-28T07:49:07.0014336Z   modrinth-id: xennote
2024-11-28T07:49:07.0014891Z   modrinth-token: ***
2024-11-28T07:49:07.0015367Z   github-token: ***
2024-11-28T07:49:07.0015739Z   version-type: beta
2024-11-28T07:49:07.0016058Z   modrinth-featured: ${undefined}
2024-11-28T07:49:07.0016440Z   modrinth-unfeature-mode: subset
2024-11-28T07:49:07.0016744Z   curseforge-id: ${undefined}
2024-11-28T07:49:07.0017075Z   curseforge-token: ${undefined}
2024-11-28T07:49:07.0017447Z   github-tag: ${undefined}
2024-11-28T07:49:07.0017780Z   github-generate-changelog: ${undefined}
2024-11-28T07:49:07.0018140Z   github-draft: ${undefined}
2024-11-28T07:49:07.0018466Z   github-prerelease: ${undefined}
2024-11-28T07:49:07.0018803Z   github-commitish: ${undefined}
2024-11-28T07:49:07.0019204Z   github-discussion: ${undefined}
2024-11-28T07:49:07.0019833Z   files: build/libs/!(*-@(dev|sources|javadoc)).jar
build/libs/*-@(dev|sources|javadoc).jar

2024-11-28T07:49:07.0020269Z   files-primary: ${undefined}
2024-11-28T07:49:07.0020634Z   files-secondary: ${undefined}
2024-11-28T07:49:07.0020967Z   name: ${undefined}
2024-11-28T07:49:07.0021240Z   version: ${undefined}
2024-11-28T07:49:07.0021589Z   changelog: ${undefined}
2024-11-28T07:49:07.0021864Z   changelog-file: ${undefined}
2024-11-28T07:49:07.0022168Z   loaders: ${undefined}
2024-11-28T07:49:07.0022522Z   game-versions: ${undefined}
2024-11-28T07:49:07.0022906Z   dependencies: ${undefined}
2024-11-28T07:49:07.0023189Z   version-resolver: ${undefined}
2024-11-28T07:49:07.0023568Z   game-version-filter: releases
2024-11-28T07:49:07.0023874Z   java: ${undefined}
2024-11-28T07:49:07.0024130Z   retry-attempts: 2
2024-11-28T07:49:07.0024499Z   retry-delay: 10000
2024-11-28T07:49:07.0024735Z   fail-mode: fail
2024-11-28T07:49:07.0025009Z   modrinth-files: ${undefined}
2024-11-28T07:49:07.0025711Z   curseforge-files: ${undefined}
2024-11-28T07:49:07.0026056Z   github-files: ${undefined}
2024-11-28T07:49:07.0026650Z   modrinth-files-primary: ${undefined}
2024-11-28T07:49:07.0027086Z   curseforge-files-primary: ${undefined}
2024-11-28T07:49:07.0027396Z   github-files-primary: ${undefined}
2024-11-28T07:49:07.0027822Z   modrinth-files-secondary: ${undefined}
2024-11-28T07:49:07.0028190Z   curseforge-files-secondary: ${undefined}
2024-11-28T07:49:07.0028551Z   github-files-secondary: ${undefined}
2024-11-28T07:49:07.0028948Z   modrinth-name: ${undefined}
2024-11-28T07:49:07.0029221Z   curseforge-name: ${undefined}
2024-11-28T07:49:07.0029542Z   github-name: ${undefined}
2024-11-28T07:49:07.0029914Z   modrinth-version: ${undefined}
2024-11-28T07:49:07.0030229Z   curseforge-version: ${undefined}
2024-11-28T07:49:07.0030546Z   github-version: ${undefined}
2024-11-28T07:49:07.0030900Z   modrinth-version-type: ${undefined}
2024-11-28T07:49:07.0031328Z   curseforge-version-type: ${undefined}
2024-11-28T07:49:07.0031661Z   github-version-type: ${undefined}
2024-11-28T07:49:07.0032062Z   modrinth-changelog: ${undefined}
2024-11-28T07:49:07.0032405Z   curseforge-changelog: ${undefined}
2024-11-28T07:49:07.0032677Z   github-changelog: ${undefined}
2024-11-28T07:49:07.0032905Z   modrinth-changelog-file: ${undefined}
2024-11-28T07:49:07.0033158Z   curseforge-changelog-file: ${undefined}
2024-11-28T07:49:07.0033407Z   github-changelog-file: ${undefined}
2024-11-28T07:49:07.0033643Z   modrinth-loaders: ${undefined}
2024-11-28T07:49:07.0033860Z   curseforge-loaders: ${undefined}
2024-11-28T07:49:07.0034086Z   github-loaders: ${undefined}
2024-11-28T07:49:07.0034312Z   modrinth-game-versions: ${undefined}
2024-11-28T07:49:07.0034553Z   curseforge-game-versions: ${undefined}
2024-11-28T07:49:07.0034802Z   github-game-versions: ${undefined}
2024-11-28T07:49:07.0035043Z   modrinth-dependencies: ${undefined}
2024-11-28T07:49:07.0035283Z   curseforge-dependencies: ${undefined}
2024-11-28T07:49:07.0035522Z   github-dependencies: ${undefined}
2024-11-28T07:49:07.0035776Z   modrinth-version-resolver: ${undefined}
2024-11-28T07:49:07.0036255Z   curseforge-version-resolver: ${undefined}
2024-11-28T07:49:07.0036532Z   github-version-resolver: ${undefined}
2024-11-28T07:49:07.0036794Z   modrinth-game-version-filter: ${undefined}
2024-11-28T07:49:07.0037070Z   curseforge-game-version-filter: ${undefined}
2024-11-28T07:49:07.0037341Z   github-game-version-filter: ${undefined}
2024-11-28T07:49:07.0037588Z   modrinth-java: ${undefined}
2024-11-28T07:49:07.0037803Z   curseforge-java: ${undefined}
2024-11-28T07:49:07.0038014Z   github-java: ${undefined}
2024-11-28T07:49:07.0038229Z   modrinth-retry-attempts: ${undefined}
2024-11-28T07:49:07.0038480Z   curseforge-retry-attempts: ${undefined}
2024-11-28T07:49:07.0038732Z   github-retry-attempts: ${undefined}
2024-11-28T07:49:07.0038970Z   modrinth-retry-delay: ${undefined}
2024-11-28T07:49:07.0039209Z   curseforge-retry-delay: ${undefined}
2024-11-28T07:49:07.0039448Z   github-retry-delay: ${undefined}
2024-11-28T07:49:07.0039679Z   modrinth-fail-mode: ${undefined}
2024-11-28T07:49:07.0039911Z   curseforge-fail-mode: ${undefined}
2024-11-28T07:49:07.0040142Z   github-fail-mode: ${undefined}
2024-11-28T07:49:07.0040349Z env:
2024-11-28T07:49:07.0040574Z   JAVA_HOME: /opt/hostedtoolcache/Java_Microsoft_jdk/21.0.2/x64
2024-11-28T07:49:07.0040952Z   JAVA_HOME_21_X64: /opt/hostedtoolcache/Java_Microsoft_jdk/21.0.2/x64
2024-11-28T07:49:07.0041261Z ##[endgroup]
2024-11-28T07:49:07.3279283Z 📤 Uploading assets to Modrinth
2024-11-28T07:49:07.3300436Z ##[error]ArgumentError: At least one game version should be specified to upload files to Modrinth. (Parameter 'request.gameVersions')
2024-11-28T07:49:07.3407950Z Post job cleanup.
2024-11-28T07:49:07.5069416Z Post job cleanup.
2024-11-28T07:49:07.6054907Z [command]/usr/bin/git version
2024-11-28T07:49:07.6096148Z git version 2.47.0
2024-11-28T07:49:07.6140977Z Temporarily overriding HOME='/home/runner/work/_temp/dfad385d-8142-4b67-b716-2dc2e377afbd' before making global git config changes
2024-11-28T07:49:07.6142488Z Adding repository directory to the temporary git global config as a safe directory
2024-11-28T07:49:07.6147035Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/xennote/xennote
2024-11-28T07:49:07.6185851Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2024-11-28T07:49:07.6221094Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2024-11-28T07:49:07.6461643Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2024-11-28T07:49:07.6485165Z http.https://github.com/.extraheader
2024-11-28T07:49:07.6496263Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2024-11-28T07:49:07.6530770Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2024-11-28T07:49:07.6912536Z Cleaning up orphan processes
2024-11-28T07:49:07.7189642Z Terminate orphan process: pid (1956) (java)
```

### Other Information

The respository: [here](/zhenlige/xennote)

**Comment by Kira-NT** [comment]
Hi! This issue is a duplicate of #121. Please refer to https://github.com/Kir-Antipov/mc-publish/issues/121#issuecomment-2155665478, where I provided a detailed explanation of the problem.

**TL;DR**: `>=1.21.2-` is not a standard syntax and is not supported by the SemVer parser I use internally. Instead, you can use syntax like `>1.21.1 <1.22` to convey the same meaning.

> Set to `[1.21.2, 1.21.3]`: `The workflow is not valid. .github/workflows/publish.yml (Line: 33, Col: 26): A sequence was not expected`

This is not on me. GitHub Workflows do not support supplying actions with non-primitive inputs like arrays or objects. If you ever need to specify versions directly as an action input, please use the syntax outlined in the README:

```yaml
game-versions: |
  1.21.2
  1.21.3
```
